src/openeo_test_suite/tests/processes/metadata/test_process_metadata.py

```diff
@@ -0,0 +1,190 @@
+import json
+import pytest
+import requests
+from openeo_test_suite.lib.process_selection import get_selected_processes
+from openeo_test_suite.lib.backend_under_test import get_backend_url
+
+
+@pytest.fixture(scope="module")
+def api_processes(request):
+    endpoint_path = "processes"
+    base_url = get_backend_url(request.config)
+    if not base_url:
+        raise ValueError("No backend URL configured")
+    if base_url.endswith("/"):
+        base_url = base_url[:-1]
+    full_endpoint_url = f"{base_url}/{endpoint_path}"
+    response = json.loads(requests.get(full_endpoint_url).content)
+    return response["processes"]
+
+
+_get_selected_processes = get_selected_processes.__wrapped__
+
+
+def get_examples():
+    """Collect process examples/tests from examples root folder containing JSON5 files."""
+    return [process for process in _get_selected_processes()]
+
+
+@pytest.mark.parametrize(
+    "expected_process",
+    [process for process in get_examples() if process.metadata != {}],
+    ids=[process.process_id for process in get_examples() if process.metadata != {}],
+)
+def test_process_metadata_functional(api_processes, expected_process):
+    """
+    Tests if the metadata of processes are correct, first tests if the process exists,
+    then tests if the parameters of processes are correct and finally tests if the return type of processes is correct.
+
+    Any process that has no metadata is skipped.
+
+    These are the functional parts of the process metadata, e.g. existence (has to be checked) the parameters and return type.
+    """
+    # Tests if the process exists
+    assert expected_process.process_id in [
+        process["id"] for process in api_processes
+    ], f"The process '{expected_process.process_id}' was not found on the backend"
+
+    # Tests if the parameters of processes are correct
```

Comment:
Parameters (and potentially also some other things) can be changed to cater for differences in the implementation status, so you may for example remove a parameter, adapt an options object in ard_* processes or change an enum in a schema for the resample methods. This is encouraged and I think we should not discourage this through the tests. As such, it might make more sense to have an optional strict mode that checks this, but by default only throw warnings for some components of the process. 

Things thay may only produce a warning could be:
- Changes in the description
- Removing parameters at the end of the list that have a default value
- Adding parameters at the end of the list that have a default value
- Changes in the parameter descriptions
- Changes in the parameter schemas
- Changes in the categories
- Additional exceptions
- Removal or Addition of examples
- Removal or Addition of links
- Removal of a process graph

Maybe also:
- Changes in the return value descriptions
- Changes in the return value schemas

Should probably not be checked as backends are always allowed to say that their implementation is deprecated of experimental. But they should not set it to false, if the source process has set them to true.
- Deprecation = true
- Experimental = true

Reply:
This was the idea for seperating non-functional and functional parts of the process metadata.

So in summary having only a subset of the parameters/return types supported will throw an error. And having a superset with default values will produce a warning only.

Any change in non-functional metadata, including categories, should only produce a warning?

I'll look into this, but it might take until next week.

Reply:
I've implemented the checks for differing lengths of parameter lists, throwing only a warning when they are not of the same length, but failing the test if default values are missing on any of the added or removed.

I have also added the checks for the deprecation and experimental flags as described.

The readme.md now also clearly states the purpose of non-functional vs. functional tests so that failure of a non-functional test is treated as a warning, but in turn I have refrained from throwing a warning for each failure.